advanced_filters/forms.py

```diff
@@ -96,12 +96,10 @@ def _build_query_dict(self, formdata=None):
         if self.is_valid() and formdata is None:
             formdata = self.cleaned_data
         key = "{field}__{operator}".format(**formdata)
-        if formdata['operator'] == "isnull":
-            return {key: None}
-        elif formdata['operator'] == "istrue":
-            return {formdata['field']: True}
-        elif formdata['operator'] == "isfalse":
-            return {formdata['field']: False}
+
+        if formdata['operator'] in ["isnull", "istrue", "isfalse"]:
+            return {key: True if str(formdata['value']).lower() in ['1', 'true'] else False}
```

Comment:
Is this functionally identical to before, or did something change here? Or is the change that `isnull` operator should return `{use key: True}`?

From a readability perspective, I think  I prefer the previous code

Reply:
Yes, this is the most important part. Otherwise it returns ``'True'`` value (string) instead of ``True`` (boolean), which is incorrect and produces false SQL statement. That's the reason why I iterate only boolean operators ("isnull", "istrue", "isfalse") and not all of the ``AdvancedFilterQueryForm.OPERATORS``.

Except of that. It allows to combine values like following:

```
isnull: True
isnull: False
istrue: True
istrue: False
isfalse: True
isfalse: False
```

which can be useful as well.